api/management-model/src/test/java/org/apache/polaris/core/admin/model/CatalogSerializationTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.core.admin.model;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class CatalogSerializationTest {
+
+  private ObjectMapper mapper;
+  private static final String TEST_LOCATION = "s3://test/";
+  private static final String TEST_CATALOG_NAME = "test-catalog";
+  private static final String TEST_ROLE_ARN = "arn:aws:iam::123456789012:role/test-role";
+
+  @BeforeEach
+  public void setUp() {
+    mapper = new ObjectMapper();
+    // Mapper to use the concrete Catalog class
+    mapper.addMixIn(PolarisCatalog.class, Catalog.class);
+  }
+
+  private Catalog verifyRoundTrip(Catalog original) throws JsonProcessingException {
+    String json = mapper.writeValueAsString(original);
+    int typeFieldCount = json.split("\"type\"").length - 1;
+    assertThat(typeFieldCount)
```

Comment:
This is too complicated, IMHO. The JSON string form of `Catalog` is stable. It will not change from one execution to another. I'd prefer to assert the exact string JSON representation in a test method dedicated to this case and leave this assertion out of the "round trip" tests.